test/integration/scheduler/scoring/priorities_test.go

```diff
@@ -756,10 +761,139 @@ func TestPodAffinityScoring(t *testing.T) {
+			name:           "no taints or tolerations",
+			podTolerations: []v1.Toleration{},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-1").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+						"nvidia.com/gpu":  "4",
+					}).Obj(),
+				st.MakeNode().Name("node-2").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+						"nvidia.com/gpu":  "4",
+					}).Obj(),
+			},
+			expectedNodesName: []string{"node-1", "node-2"},
+		},
+		{
+			name: "pod with toleration for node's taint",
+			podTolerations: []v1.Toleration{
+				{
+					Key:      "example-key",
+					Operator: v1.TolerationOpEqual,
+					Value:    "example-value",
+					Effect:   v1.TaintEffectPreferNoSchedule,
+				},
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-1").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+					}).Taints([]v1.Taint{
+					{
+						Key:    "example-key",
+						Value:  "example-value",
+						Effect: v1.TaintEffectPreferNoSchedule,
+					},
+				}).Obj(),
+				st.MakeNode().Name("node-2").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+					}).Obj(),
+			},
+			expectedNodesName: []string{"node-1", "node-2"},
+		},
+		{
+			name: "pod without toleration for node's taint",
+			podTolerations: []v1.Toleration{
+				{
+					Key:      "other-key",
+					Operator: v1.TolerationOpEqual,
+					Value:    "other-value",
+					Effect:   v1.TaintEffectPreferNoSchedule,
+				},
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-1").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+					}).Taints([]v1.Taint{
+					{
+						Key:    "example-key",
+						Value:  "example-value",
+						Effect: v1.TaintEffectPreferNoSchedule,
+					},
+				}).Obj(),
+				st.MakeNode().Name("node-2").Capacity(
+					map[v1.ResourceName]string{
+						v1.ResourceCPU:    "8",
+						v1.ResourceMemory: "16G",
+					}).Obj(),
+			},
+			expectedNodesName: []string{"node-2"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testCtx := initTestSchedulerForScoringTests(t, tainttoleration.Name, tainttoleration.Name)
+
+			for _, n := range tt.nodes {
+				if _, err := createNode(testCtx.ClientSet, n); err != nil {
+					t.Fatalf("failed to create node: %v", err)
+				}
+			}
+			// Create a dummy pod with the given tolerations
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: testCtx.NS.Name,
+					Name:      fmt.Sprintf("test-pod-%v", rand.IntN(10)),
+				},
+				Spec: v1.PodSpec{
+					Tolerations: tt.podTolerations,
+					Containers: []v1.Container{
+						{
+							Name:  "test-container",
+							Image: imageutils.GetPauseImageName(),
+						},
+					},
+				},
+			}
+
+			pod, err := runPausePod(testCtx.ClientSet, pod)
```

Comment:
You could change to:
```suggestion
			pod, err := runPausePod(testCtx.ClientSet, initPausePod(&testutils.PausePodConfig{
				Name:      fmt.Sprintf("test-pod-%v", rand.IntN(10)),
				Namespace: testCtx.NS.Name,
				Tolerations: tt.podTolerations,
			}))
```

Reply:
done